include/nbl/asset/ICPURayTracingPipeline.h

```diff
@@ -138,6 +138,23 @@ class ICPURayTracingPipeline final : public ICPUPipeline<IRayTracingPipeline<ICP
             return dependants;
         }
 
+        inline virtual void visitDependentsImpl(std::function<bool(const IAsset*)> visit) const override
+        {
+            core::unordered_set<const IAsset*> dependants;
+            const auto visitOnce = [&](const IAsset* dep) -> bool {
+                auto [iter, inserted] = dependants.insert(dep);
+                if (inserted) return visit(dep);
+                return true;
+            };
+            visitOnce(m_raygen.shader.get());
+            for (const auto& missInfo : self->m_misses) visitOnce(missInfo.shader.get());
+            for (const auto& anyHitInfo : self->m_hitGroups.anyHits) visitOnce(anyHitInfo.shader.get());
+            for (const auto& closestHitInfo : self->m_hitGroups.closestHits) visitOnce(closestHitInfo.shader.get());
+            for (const auto& intersectionInfo : self->m_hitGroups.intersections) visitOnce(intersectionInfo.shader.get());
+            for (const auto& callableInfo : self->m_callables) visitOnce(callableInfo.shader.get());
```

Comment:
or allow multiple visits like TLAS does

Reply:
you should allow multiple visits to a dependant
if some algo doesn't want that, they can implement that check in the lambda

Reply:
definitely allow multiple visits, if someone doesn't want that let them handle it in the `visit` function